lib/spack/spack/modules.py

```diff
@@ -275,48 +277,23 @@ def naming_scheme(self):
         return naming_scheme
 
     @property
-    def tokens(self):
-        """Tokens that can be substituted in environment variable values
-        and naming schemes
-        """
-        tokens = {
-            'name': self.spec.name,
-            'version': self.spec.version,
-            'compiler': self.spec.compiler,
-            'prefix': self.spec.package.prefix
-        }
-        return tokens
-
-    @property
-    def upper_tokens(self):
-        """Tokens that can be substituted in environment variable names"""
-        upper_tokens = {
-            'name': self.spec.name.replace('-', '_').upper()
-        }
-        return upper_tokens
-
-    @property
     def use_name(self):
         """
         Subclasses should implement this to return the name the module command
         uses to refer to the package.
         """
-        naming_tokens = self.tokens
-        naming_scheme = self.naming_scheme
-        name = naming_scheme.format(**naming_tokens)
+        name = self.spec.format(self.naming_scheme)
         # Not everybody is working on linux...
         parts = name.split('/')
         name = join_path(*parts)
         # Add optional suffixes based on constraints
         configuration, _ = parse_config_options(self)
         suffixes = [name]
+        #TODO: leave this for now - just replace the default naming
+        #scheme with a spec.format-compatible one
```

Comment:
Just in case : I believe hashes should always come last in the name, i.e. : 
```
<rest-of-the-name>-<suffixes>-<hash>
```
and not : 
```
<rest-of-the-name>-<hash>-<suffixes>
```